src/liballoc/raw_vec.rs

```diff
@@ -131,17 +142,30 @@ impl<T> RawVec<T, Global> {
     /// `RawVec` with capacity `0`. If `T` is zero-sized, then it makes a
     /// `RawVec` with capacity `usize::MAX`. Useful for implementing
     /// delayed allocation.
+    #[cfg(not(bootstrap))]
     pub const fn new() -> Self {
         // FIXME(Centril): Reintegrate this with `fn new_in` when we can.
 
-        // `!0` is `usize::MAX`. This branch should be stripped at compile time.
-        // FIXME(mark-i-m): use this line when `if`s are allowed in `const`:
-        //let cap = if mem::size_of::<T>() == 0 { !0 } else { 0 };
+        let cap = if mem::size_of::<T>() == 0 { !0 } else { 0 };
```

Comment:
Could you move the `cfg(bootstrap)` stuff *inside* the function definition? I think `cfg_if` would work for this, or you could find a way to write it by hand.

Reply:
Done. `cfg_if` wouldn't have worked I think because the `then` would still need to have the `const_if_match` feature, which doesn't exist yet for the first stage.

Reply:
`cfg_if` [will in fact work](https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=0bec4ab0c2451b50d232bd791694f03a). My concern was more about adding an extra dependency.